pkg/virt-handler/hotplug-disk/mount.go

```diff
@@ -613,12 +616,21 @@ func (m *volumeMounter) getSourcePodFilePath(sourceUID types.UID, vmi *v1.Virtua
 	}
 	for _, findmnt := range findmounts {
 		if filepath.Base(findmnt.Target) == volume {
-			source := findmnt.GetSource()
+			source := findmnt.GetSourcePath()
 			isBlock, _ := isBlockDevice(filepath.Join(util.HostRootMount, source))
 			if _, err := os.Stat(filepath.Join(util.HostRootMount, source)); os.IsNotExist(err) || isBlock {
 				// file not found, or block device, or directory check if we can find the mount.
 				deviceFindMnt, err := LookupFindmntInfoByDevice(source)
 				if err != nil {
+					// Try the device found from the source
+					deviceFindMnt, err = LookupFindmntInfoByDevice(findmnt.GetSourceDevice())
+					if err != nil {
+						return "", err
+					}
+					// Check if the path was relative to the device.
+					if _, err := os.Stat(filepath.Join(util.HostRootMount, source)); err != nil {
+						return filepath.Join(deviceFindMnt[0].Target, source), nil
```

Comment:
Why it only checks if `hostRoot/source` exists and does not check `deviceFindMnt[0].Target/source`.

Also this code is run when file not found or  storage `isBlock`? Is the logic the same for both?

Reply:
```go
if _, err := os.Stat(filepath.Join(util.HostRootMount, source)); os.IsNotExist(err) || isBlock 
```
Checks if the returned path doesn't exists, or if the returned path does exist, but is a block device (isBlock)
If either is true, then we call deviceFindMnt with that path. Basically figuring out where this path is mounted (if it is a block device for instance). If found return the target (mount path). If not found we try again but this time with the SourceDevice part of the findmnt information. The arguments passed to findmtn return 'source' in the following format:
```
/dev/device[/path/relative/to/device]
```
findmnt.GetSourceDevice() returns the /dev/device part
findmnt.GetSourcePath() returns /path/relative/to/device part

Once I get to
```go
if _, err := os.Stat(filepath.Join(util.HostRootMount, source)); err != nil {
```
I know that deviceFindMnt[0].Target is the path the device is mounted on relative to the boot partition. And I know that source is the path relative to the device. So the two combined will be the path I am looking for. I can check if that path exists, but if it doesn't I have no recourse to do anything but return an error, and the caller of this function is going to try and do stuff with the path immeidately as well, and error if it doesn't exist then.

Reply:
Thanks, this explains it. I just wonder if this can be simplified.